rites/roman1969/__tests__/hong-kong-calendar.test.ts

```diff
@@ -0,0 +1,173 @@
+import { HongKong_En } from '@dist/rite-roman1969/bundles/hong-kong';
+import { LiturgicalCalendar, Romcal } from '@src/rite-roman1969';
+
+import { Precedences } from '../src/constants/precedences';
+import { Ranks } from '../src/constants/ranks';
+
+describe('Testing Hong Kong calendar specific celebrations', () => {
+  const romcal = new Romcal({ localizedCalendar: HongKong_En });
+
+  let calendar2024: LiturgicalCalendar;
+  let calendar2025: LiturgicalCalendar;
+
+  beforeAll(async () => {
+    calendar2024 = await romcal.generateCalendar(2024);
+    calendar2025 = await romcal.generateCalendar(2025);
+  });
+
+  describe('Hong Kong Saints and Blesseds', () => {
+    test('St. Joseph Freinademetz should be celebrated on January 29', () => {
+      const jan29 = calendar2024['2024-01-29'];
+      const freinademetz = jan29?.find((day) => day.id === 'joseph_freinademetz_priest');
+
+      expect(freinademetz).toBeDefined();
+      expect(freinademetz?.name).toBe('Saint Joseph Freinademetz, Priest');
+      expect(freinademetz?.rank).toBe(Ranks.Memorial);
+      expect(freinademetz?.precedence).toBe(Precedences.ProperMemorial_11b);
+    });
+
+    test('Blessed Gabriele Maria Allegra should be celebrated on January 30', () => {
+      const jan30 = calendar2024['2024-01-30'];
+      const allegra = jan30?.find((day) => day.id === 'gabriele_maria_allegra_priest');
+
+      expect(allegra).toBeDefined();
+      expect(allegra?.name).toBe('Blessed Gabriele Maria Allegra, Priest');
+      expect(allegra?.rank).toBe(Ranks.Memorial);
+      expect(allegra?.precedence).toBe(Precedences.ProperMemorial_11b);
+    });
+
+    test('Sts. Aloysius Versiglia and Callistus Caravario should be celebrated on February 25', () => {
+      // 2024 is Leap Year. Feb 25 2024 is Sunday.
```

Comment:
We could add a test that in 2024, the celebration is missing and that Sunday is celebration instead.